src/classes/QueryPredicate.cls

```diff
@@ -0,0 +1,60 @@
+/*************************************************************************************************
+* This file is part of the Nebula Framework project, released under the MIT License.             *
+* See LICENSE file or go to https://github.com/jongpie/NebulaFramework for full license details. *
+*************************************************************************************************/
+public class QueryPredicate {
```

Comment:
Cool!  Let me know what else you think.  Now that we have query predicates, any time we create a new object, it's easy for us to spin up a new repo with the virtual SObjectRepository, a short list of fields, and our factory class that handles actual object creation.  We did that today with a new joint object and it took all of a minute.  Even if sometimes we're using the repo just for the crud / dml operations, you can definitely see the progression as we move towards defining the boundaries - and objects that were previously just being inserted begin to take on new logic as we find the need to check inserted / updated values.  Honestly, between the trigger framework and the SObjectRepository (to say nothing of the mocks, which have sped up testing time to mere seconds), it's been fun to imagine and then easily implement the business logic we're building out.  

Looking forward to seeing your take on the querying parent objects, as well!